diff_drive_controller/src/diff_drive_controller.cpp

```diff
@@ -576,7 +777,163 @@ CallbackReturn DiffDriveController::configure_side(
+{
+  if (!(lookup_wheel_separation || lookup_wheel_radius)) {
+    // Short-circuit in case we don't need to look up anything, so we don't have to parse the URDF
+    return true;
+  }
+
+  // Parse robot description
+  const std::string model_param_name = "robot_description";
+  std::string robot_model_str;
+  if (!lifecycle_node_->get_parameter(model_param_name, robot_model_str)) {
+    RCLCPP_ERROR(
+      lifecycle_node_->get_logger(), "Robot description couldn't be retrieved from param server.");
+    return false;
+  }
+
+  urdf::ModelInterfaceSharedPtr model(urdf::parseURDF(robot_model_str));
+
+  urdf::JointConstSharedPtr left_wheel_joint(model->getJoint(left_wheel_name));
+  urdf::JointConstSharedPtr right_wheel_joint(model->getJoint(right_wheel_name));
+
+  if (lookup_wheel_separation) {
+    // Get wheel separation
+    if (!left_wheel_joint) {
+      RCLCPP_ERROR_STREAM(
+        lifecycle_node_->get_logger(),
+        left_wheel_name << " couldn't be retrieved from model description");
+      return false;
+    }
+
+    if (!right_wheel_joint) {
+      RCLCPP_ERROR_STREAM(
+        lifecycle_node_->get_logger(),
+        right_wheel_name << " couldn't be retrieved from model description");
+      return false;
+    }
+
+    RCLCPP_INFO_STREAM(
+      lifecycle_node_->get_logger(),
+      "left wheel to origin: " << left_wheel_joint->parent_to_joint_origin_transform.position.x
+                               << ","
+                               << left_wheel_joint->parent_to_joint_origin_transform.position.y
+                               << ", "
+                               << left_wheel_joint->parent_to_joint_origin_transform.position.z);
+    RCLCPP_INFO_STREAM(
+      lifecycle_node_->get_logger(),
+      "right wheel to origin: " << right_wheel_joint->parent_to_joint_origin_transform.position.x
+                                << ","
+                                << right_wheel_joint->parent_to_joint_origin_transform.position.y
+                                << ", "
+                                << right_wheel_joint->parent_to_joint_origin_transform.position.z);
+
+    wheel_params_.separation = urdf_util::euclidean_of_vectors(
+      left_wheel_joint->parent_to_joint_origin_transform.position,
+      right_wheel_joint->parent_to_joint_origin_transform.position);
+  }
+
+  if (lookup_wheel_radius) {
+    // Get wheel radius
+    if (!urdf_util::get_wheel_radius(
+          model->getLink(left_wheel_joint->child_link_name), wheel_params_.radius,
+          lifecycle_node_->get_logger())) {
+      RCLCPP_ERROR_STREAM(
+        lifecycle_node_->get_logger(), "Couldn't retrieve " << left_wheel_name << " wheel radius");
+      return false;
+    }
+  }
+
+  return true;
+}
+
+void DiffDriveController::publish_wheel_data(
+  const rclcpp::Time & time, const rclcpp::Duration & period, const Commands & curr_cmd,
+  double wheel_separation, double left_wheel_radius, double right_wheel_radius)
```

Comment:
It's necessary to compute the velocity before the speed limiter is applied since the objective is to compute the desired velocity. I believe there is actually a bug on the ROS1 branch where it incorrectly does the calculation after.